### PR TITLE
DEP Silverstripe CMS 5 compatibility

### PR DESCRIPTION
**Pull request by GuySartorelli**
Replacement for https://github.com/tractorcow-farm/silverstripe-fluent/pull/777
Rebased on master. This is the same code that is in the beta, but with references to `silverstripe/silverstripe-fluent` reverted back to `tractorcow/silverstripe-fluent`

CI is green in our fork: https://github.com/creative-commoners/silverstripe-fluent/actions/runs/4190234023

## After merging
Ideally this would be tagged `7.0.0-beta1` - we can then (potentially) update our recipes to point back here instead of the fork - but even if we don't, it gives people the option of installing the beta of this module with the correct composer name.

## Parent issue
- https://github.com/tractorcow-farm/silverstripe-fluent/issues/776

### CODE REVIEW

**Comment by GuySartorelli** [comment]
Added one last commit to remove deprecated methods which were deprecated in fluent 5 and aren't called in any of the code in 6 or 7.

**Comment by GuySartorelli** [comment]
Not sure why scrutinizer is angry (can't install npm 18 for some reason?), but @tractorcow unless you've been regularly checking it I'd recommend either removing it entirely or, as a less drastic approach, accept the scrutinizer failure for this PR and handle that separately.

File: package.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
   "name": "tractorcow-fluent",
-  "version": "4.0.0",
   "description": "Fluent localisation module for SilverStrip CMS",
   "engines": {
-    "node": "^10.x"
+    "node": "^18.x"
   },
   "scripts": {
-    "build": "NODE_ENV=production webpack -p --bail --progress",
+    "build": "yarn && yarn lint && rm -rf client/dist/* && NODE_ENV=production webpack --mode production --bail --progress",
```
Comment (mfendeksilverstripe) [inline package.json:8]: Why is the `rm` command present? I don't see this in any other modules, for example https://github.com/silverstripe/silverstripe-linkfield/blob/1/package.json#L20

File: tests/php/Extension/FluentExtensionTest.php [hunk 2]
```diff
@@ -289,12 +289,16 @@ public function testLocalisedCopyClassNameChange(): void
...
      */
-    public function testLocalisedFieldsCanBeSorted($locale, array $sortArgs, $expected)
+    public function testLocalisedFieldsCanBeSorted($locale, array $sortArgs, $expected, $useOrderBy = false)
     {
-        FluentState::singleton()->withState(function (FluentState $newState) use ($locale, $sortArgs, $expected) {
+        FluentState::singleton()->withState(function (FluentState $newState) use ($locale, $sortArgs, $expected, $useOrderBy) {
             $newState->setLocale($locale);
 
-            $records = LocalisedParent::get()->sort(...$sortArgs);
+            if ($useOrderBy) {
```
Comment (mfendeksilverstripe) [inline tests/php/Extension/FluentExtensionTest.php:297]: [nitpick] Use ternary operator.

**Comment by mfendeksilverstripe** [comment]
Left some comments but the changes look good so far 👍

File: package.json (hunk 1 above)
Comment (GuySartorelli) [reply package.json:8]: linkfield isn't a supported module.
Look at any supported module's CMS 5 compatible version, e.g. https://github.com/silverstripe/silverstripe-subsites/blob/3.0/package.json#L4

It's to protect against old files from earlier builds sticking around (e.g. as a result of webpack copy), or from people putting things in dist manually. We should always be able to blow away dist and rebuild and nothing will be lost - this ensures that's always the case.

File: tests/php/Extension/FluentExtensionTest.php (hunk 2 above)
Comment (GuySartorelli) [reply tests/php/Extension/FluentExtensionTest.php:297]: That's subjective, so unless there's some specific standard that this module adheres to which requires the ternary here I'm inclined to not make this change in this PR.

Comment (mfendeksilverstripe) [reply tests/php/Extension/FluentExtensionTest.php:297]: Agree, hence the nitpick flag. Happy to keep it as is.

**Comment by maxime-rainville** [comment]
Just approved the build to run.

**Review by tractorcow — APPROVED** [review]

**Comment by tractorcow** [comment]
Merge at will. let me know if you need help releasing /etc.

**Comment by maxime-rainville** [comment]
I just:
- merged the PR, 
- created  `7.0` and `7` branch from master,
- Tag a 7.0.0-beta1 from 7.0.

**Comment by tractorcow** [comment]
Nice job :D

**Pull request merged**